src/app/components/pages/Groups.tsx

```diff
@@ -526,6 +525,120 @@ export const Groups = ({user}: {user: RegisteredUserDTO}) => {
+                                    }
+                                </div>
+                            </div>
+                            : <div key={g.id} className="group-item p-2" data-testid={"group-item"}>
+                                <div className="d-flex justify-content-between align-items-center group-name-buttons">
+                                    <Link to={`/groups#${g.id}`} title={isStaff(user) ? `Group id: ${g.id}` : undefined} data-testid={"select-group"} className="text-start px-1 py-1 group-name d-flex flex-fill">
+                                        {g.groupName}
+                                    </Link>
+                                    {showArchived &&
+                                        <button onClick={(e) => {e.stopPropagation(); confirmDeleteGroup(g);}}
+                                            aria-label="Delete group" className={classNames("ms-1", siteSpecific("icon-close", "bin-icon"))} title={"Delete group"}/>
+                                    }
+                                </div>
+                            </div>
+                    )
+                    : <div className={"group-item p-2"}>No {showArchived ? "archived" : "active"} groups</div>
+                }
+            </div>
+        </CardBody>
+    </Card>;
+};
+
+const stateToProps = (_state: AppState, props: any) => {
+    const {location: {hash}} = props;
+    return {hashAnchor: hash?.slice(1) ?? null};
+};
+
+const GroupsComponent = ({user, hashAnchor}: {user: RegisteredUserDTO, hashAnchor: number}) => {
+    const dispatch = useAppDispatch();
+    const deviceSize = useDeviceSize();
+
+    const [showArchived, setShowArchived] = useState(false);
+    const groupQuery = useGetGroupsQuery(showArchived);
+    const { currentData: groups, isLoading, isFetching } = groupQuery;
+    const otherGroups = useGetGroupsQuery(!showArchived);
+
+    const allGroups = [...(groups ?? []) , ...(otherGroups.currentData ?? [])];
+
+    const [createGroup] = useCreateGroupMutation();
+
+    const [selectedGroupId, setSelectedGroupId] = useState<number | undefined>();
+    useEffect(() => {
+        // @ts-ignore
+        const tab: number = hashAnchor && parseInt(hashAnchor);
+        setSelectedGroupId(tab);
+    }, [hashAnchor]);
+    const selectedGroup = (isLoading || isFetching) ? undefined : groups?.find(g => g.id === selectedGroupId);
+
+    const createNewGroup: (newGroupName: string) => Promise<boolean> = async (newGroupName: string) => {
+        setShowArchived(false);
+        return createGroup(newGroupName).then(async (result) => {
+            if (mutationSucceeded(result)) {
+                const group = result.data;
+                if (!group.id) return false;
+                dispatch(showGroupInvitationModal({group, user, firstTime: true, backToCreateGroup: () => setSelectedGroupId(undefined)}));
```

Comment:
This modal isn't showing up for me. I've had a quick look but nothing is jumping at me. Any ideas?

Reply:
Good catch! This seems to be happening because of the `setSelectedGroupId` immediately after the modal is dispatched, which triggers a `useEffect` - the problematic line is `window.location.hash = selectedGroupId.toString()`, which causes an `ACTIVE_MODAL_CLOSE`. I have a fix [here](https://github.com/isaacphysics/isaac-react-app/compare/redesign/page/groups...groups-modal-bugfix?expand=1) which just moves the modal dispatch to after the hash is changed, but there might be a neater way to deal with this - any thoughts would be appreciated :)

Reply:
Ahhh, nice. Changing the hash causes a soft reload which removes the modal. We could just set the hash early, so the `useEffect` doesn't change anything:
```suggestion
                // setting the hash early here prevents the useEffect from changing the hash and causing a reload after the modal is dispatched, which would remove it.
                window.location.hash = group.id.toString(); 
                dispatch(showGroupInvitationModal({group, user, firstTime: true, backToCreateGroup: () => setSelectedGroupId(undefined)}));
```